pennylane/ops/qubit/qchem_ops.py

```diff
@@ -320,10 +331,20 @@ def circuit(phi):
     grad_method = "A"
     grad_recipe = four_term_grad_recipe
 
-    G = np.zeros((16, 16), dtype=np.complex64)
-    G[3, 12] = -1j  # 3 (dec) = 0011 (bin)
-    G[12, 3] = 1j  # 12 (dec) = 1100 (bin)
-    generator = [G, -1 / 2]
+    def generator(self):
+        w0, w1, w2, w3 = self.wires
+        coeffs = [0.0625, 0.0625, -0.0625, 0.0625, -0.0625, 0.0625, -0.0625, -0.0625]
+        obs = [
+            qml.PauliX(w0) @ qml.PauliX(w1) @ qml.PauliX(w2) @ qml.PauliY(w3),
+            qml.PauliX(w0) @ qml.PauliX(w1) @ qml.PauliY(w2) @ qml.PauliX(w3),
+            qml.PauliX(w0) @ qml.PauliY(w1) @ qml.PauliX(w2) @ qml.PauliX(w3),
+            qml.PauliX(w0) @ qml.PauliY(w1) @ qml.PauliY(w2) @ qml.PauliY(w3),
+            qml.PauliY(w0) @ qml.PauliX(w1) @ qml.PauliX(w2) @ qml.PauliX(w3),
+            qml.PauliY(w0) @ qml.PauliX(w1) @ qml.PauliY(w2) @ qml.PauliY(w3),
+            qml.PauliY(w0) @ qml.PauliY(w1) @ qml.PauliX(w2) @ qml.PauliY(w3),
+            qml.PauliY(w0) @ qml.PauliY(w1) @ qml.PauliY(w2) @ qml.PauliX(w3),
+        ]
+        return qml.Hamiltonian(coeffs, obs)
```

Comment:
Will such large generators lead to overheads in preprocessing, in particular on simulators where we are not relying on fully decomposing into elementary gates anyways? For example, should the neat, short definition of the matrix be hardcoded somewhere as well?

Reply:
Yes. This was the point at which I had the same thought as you and swapped to hardcoding the matrices below!

A part of me wonders though if the structure of knowing the terms is useful in other applications; e.g., hardware. In particular, as the size of the generator grows, it will become intractable to measure in their basis via numerical eigenvector decomposition; the terms instead help encode the diagonalization.

Reply:
Good thinking!

Reply:
I agree! I think we might want both to be available and the backend to know what's best for itself :)